packages/admin-panel/src/VizBuilderApp/components/PreviewOptions/DateRangeField.jsx

```diff
@@ -40,7 +41,30 @@ export const DateRangeField = () => {
     }
   }, [defaultStartDate, startDate, defaultEndDate, endDate]);
 
-  const convertDateToIsoString = date => (date ? date.toISOString() : null);
+  /**
+   * When you select a date in the date picker (either with the interactive calendar or by inserting
+   * a valid date string), the picker actually selects a specific moment in time on that date with
+   * the normal, millisecond granularity of a Date object. The time of day it selects is the current
+   * local time on the system where VizBuilder is running.
+   *
+   * When inputting the date, the date picker interprets it in the user’s time zone. Under the hood,
+   * this is converted to UTC. But since we only need day-level granularity, this timezone
+   * conversion can make the date picker select a date different from what the user input (±1 day)
+   * when {@link converDateToIsoString} discards the timestamp.
+   *
+   * This helper function accounts for that discrepancy.
+   *
+   * @param date A valid date object
+   * @returns {Date}
+   */
+  const shiftEpoch = date =>
+    new Date(date.setMinutes(date.getMinutes() - date.getTimezoneOffset()));
```

Comment:
@alexd-bes This is the hacky fix I mentioned in stand-up this morning. Easy enough to understand, but you can imagine why I’m not happy about it 😅

Andrew went to the trouble of testing at multiple times throughout the day and it does seem to work, though; so it’s currently marked as **test passed**

Reply:
I think this is probably fine - as you say, it's a very specific case and it's an old install of datepicker which may get rejigged anyway, so if it works it's all good